### PR TITLE
Can select one or more cards on index grid

### PR DESCRIPTION
**Pull request by burieberry**
- use ElementTracker for overlays
- remove opening card action out of cards-grid component, that responsibility belongs to the overlay
- no longer need to have the card-author pass down ...attributes to get the overlays on linked cards
- simplify overlay offset, no longer need to adjust for padding. the overlay will cover the entire card.
- add test for selection on cards-grid card

- Note: There's a scroll issue (can't scroll the card grid when mouse is hovering on an overlay), and selected card continues to have the overlay visible when scrolled outside of the container. We'll also eventually need more fine-grained control over what shows up on an overlay. I'll create tickets for these.

grid-selection

### CODE REVIEW

File: packages/base/cards-grid.gts [hunk 1]
```diff
@@ -73,13 +109,18 @@ class Isolated extends Component<typeof CardsGrid> {
...
+      button.card-catalog-item__hover-button { width: 30px; height: 30px; }
+    </style>
   </template>
 
   @tracked request?: {
     instances: Card[];
     isLoading: boolean;
   };
 
+  @tracked selectedCards: Card[] = [];
```
Comment (FadhlanR) [inline packages/base/cards-grid.gts]: Can we use `TrackedArray` here?

File: packages/base/cards-grid.gts [hunk 2]
```diff
@@ -103,13 +144,37 @@ class Isolated extends Component<typeof CardsGrid> {
...
+  get isOperatorMode() {
+    let component = this.args.context?.renderedIn;
+    return component?.constructor?.name === 'OperatorModeStackItem';
+  }
+
+  @action toggleSelect(card: Card) {
+    if (this.isSelected(card.id)) {
+      this.selectedCards = this.selectedCards.filter((c) => c.id !== card.id);
+    } else {
+      this.selectedCards = [...this.selectedCards, card];
```
Comment (FadhlanR) [inline packages/base/cards-grid.gts]: with `TrackedArray` we don't need to reinitialize `selectedCards`.

File: packages/base/cards-grid.gts [hunk 3]
```diff
@@ -103,13 +144,37 @@ class Isolated extends Component<typeof CardsGrid> {
     });
   });
 
+  get isOperatorMode() {
+    let component = this.args.context?.renderedIn;
+    return component?.constructor?.name === 'OperatorModeStackItem';
```
Comment (lukemelia) [inline packages/base/cards-grid.gts]: This seems fragile to refactoring.

File: packages/base/cards-grid.gts [hunk 4]
```diff
@@ -103,13 +144,37 @@ class Isolated extends Component<typeof CardsGrid> {
...
+    } else {
+      this.selectedCards = [...this.selectedCards, card];
+    }
+  }
+
+  @action isSelected(id: string) {
+    if (this.selectedCards.find((card) => card.id === id)) {
+      return true;
+    }
+    return false;
```
Comment (lukemelia) [inline packages/base/cards-grid.gts]:
```suggestion
    return this.selectedCards.some((card) => card.id === id);
```

**Review by lukemelia — CHANGES_REQUESTED** [review]

File: packages/base/cards-grid.gts (hunk 3 above)
Comment (burieberry) [reply packages/base/cards-grid.gts]: Any suggestions? If we're moving selection to the operator mode, then we won't need this check here. I can try to move selection as part of this PR

File: packages/boxel-ui/addon/components/actions-overlay/index.gts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    selectedCards: TrackedArray<any>;
+    toggleSelect: (card: any) => void;
+    viewCard: (card: any) => void;
+  };
+  Blocks: {
+    default: [];
+  };
+}
+export default class ActionOverlay extends Component<Signature> {
+  refreshLoopStartedAt: number | null = null;
```
Comment (jurgenwerk) [inline packages/boxel-ui/addon/components/actions-overlay/index.gts]: I don't see these two properties being used

File: packages/boxel-ui/addon/components/actions-overlay/index.gts [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+  refreshLoopTimeout: number | null = null;
+
+  <template>
+    {{#let @card as |card|}}
```
Comment (jurgenwerk) [inline packages/boxel-ui/addon/components/actions-overlay/index.gts]: How come this block is needed? Can we just use `@card` in the code below?

Comment (burieberry) [reply packages/boxel-ui/addon/components/actions-overlay/index.gts]: Sorry this PR went back to "in progress/draft" state. I was lazy to replace while just trying stuff out before fully committing to this implementation. I'll do the clean up

**Review by lukemelia — APPROVED** [review]

**Pull request merged**